cmd/zt_credentialUtil_test.go

```diff
@@ -34,46 +34,50 @@ var _ = chk.Suite(&credentialUtilSuite{})
-		{common.ECredentialType.S3AccessKey(), "http://something.s3.cn-north-1.amazonaws.com.cn", "", true},
-		{common.ECredentialType.S3AccessKey(), "http://s3.eu-central-1.amazonaws.com", "", true},
-		{common.ECredentialType.S3AccessKey(), "http://s3.cn-north-1.amazonaws.com.cn", "", true},
-		{common.ECredentialType.S3AccessKey(), "http://s3.amazonaws.com", "", true},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://myaccount.blob.core.windows.net", "", true},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://myaccount.blob.core.chinacloudapi.cn", "", true},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://myaccount.blob.core.cloudapi.de", "", true},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://myaccount.blob.core.core.usgovcloudapi.net", "", true},
+		{common.ECredentialType.SharedKey(), common.ELocation.BlobFS(), "http://myaccount.dfs.core.windows.net", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://something.s3.eu-central-1.amazonaws.com", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://something.s3.cn-north-1.amazonaws.com.cn", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://s3.eu-central-1.amazonaws.com", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://s3.cn-north-1.amazonaws.com.cn", "", true},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://s3.amazonaws.com", "", true},
 
 		// These should fail (they are not storage)
-		{common.ECredentialType.OAuthToken(), "http://somethingelseinazure.windows.net", "", false},
-		{common.ECredentialType.S3AccessKey(), "http://somethingelseinaws.amazonaws.com", "", false},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://somethingelseinazure.windows.net", "", false},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://somethingelseinaws.amazonaws.com", "", false},
 
 		// As should these (they are nothing to do with the expected URLs)
-		{common.ECredentialType.OAuthToken(), "http://example.com", "", false},
-		{common.ECredentialType.S3AccessKey(), "http://example.com", "", false},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://example.com", "", false},
+		{common.ECredentialType.S3AccessKey(), common.ELocation.S3(), "http://example.com", "", false},
+		// Test that we don't want to send an S3 access key to a blob resource type.
+		{common.ECredentialType.S3AccessKey(), common.ELocation.Blob(), "http://example.com", "", false},
 
 		// But the same Azure one should pass if the user opts in to them (we don't support any similar override for S3)
-		{common.ECredentialType.OAuthToken(), "http://example.com", "*.foo.com;*.example.com", false},
+		{common.ECredentialType.OAuthToken(), common.ELocation.Blob(), "http://example.com", "*.foo.com;*.example.com", false},
```

Comment:
I just noticed something - this test has been wrong all along!  When I wrote it, I was supposed to have "true" at the end.  I've just pushed a small commit to fix it.  URL changes to abc.example.com so that the dot before example is there (which is expected by the *.example.com).